### PR TITLE
refactor: Explicit `checkPathLength` and `isValid` intersection check

### PR DESCRIPTION
**Pull request by andiwand**
- Rename `checkIntersection` to `checkPathLength`
  - Remove templating
- Add `isValid` to intersection objects
  - Will remove the boolean conversion in the future

### CODE REVIEW

File: Core/include/Acts/Utilities/Intersection.hpp [hunk 1]
```diff
@@ -60,9 +60,11 @@ class Intersection {
       : m_position(position), m_pathLength(pathLength), m_status(status) {}
 
   /// Returns whether the intersection was successful or not
-  constexpr explicit operator bool() const {
-    return m_status != Status::missed;
-  }
+  /// @deprecated
+  constexpr explicit operator bool() const { return isValid(); }
```
Comment (benjaminhuth) [inline Core/include/Acts/Utilities/Intersection.hpp:64]:
```suggestion
  [[deprecated("Use isValid() instead")]] constexpr explicit operator bool() const { return isValid(); }
```

File: Core/include/Acts/Utilities/Intersection.hpp [hunk 2]
```diff
@@ -141,9 +143,11 @@ class ObjectIntersection {
       : m_intersection(intersection), m_object(object), m_index(index) {}
 
   /// Returns whether the intersection was successful or not
-  constexpr explicit operator bool() const {
-    return m_intersection.operator bool();
-  }
+  /// @deprecated
+  constexpr explicit operator bool() const { return isValid(); }
```
Comment (benjaminhuth) [inline Core/include/Acts/Utilities/Intersection.hpp:147]:
```suggestion
  [[deprecated("Use isValid() instead")]] constexpr explicit operator bool() const { return isValid(); }
```

File: Core/include/Acts/Utilities/Intersection.hpp [hunk 3]
```diff
@@ -256,29 +260,25 @@ class ObjectMultiIntersection {
...
+/// @param nearLimit The minimum path length for an intersection to be considered
+/// @param farLimit The maximum path length for an intersection to be considered
 /// @param logger A optionally supplied logger which prints out a lot of infos
 ///               at VERBOSE level
-template <typename intersection_t>
-bool checkIntersection(const intersection_t& intersection, double nearLimit,
-                       double farLimit,
-                       const Logger& logger = getDummyLogger()) {
-  const double distance = intersection.pathLength();
+inline bool checkPathLength(double pathLength, double nearLimit,
```
Comment (benjaminhuth) [inline Core/include/Acts/Utilities/Intersection.hpp:271]: we could compile this into a .cpp file, not sure if performance could suffer from this though...

**Review by benjaminhuth — COMMENTED** [review]
Cool, just a few comments

**Review by benjaminhuth — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [64d681b61b723b0d3aa2f468f84e8a38c934026a]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/41516743) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7802188)!

**Pull request merged**